flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java

```diff
@@ -517,7 +527,16 @@ public void failJobDueToTaskFailure(
             registerJobStatusListener(checkpointCoordinator.createActivatorDeactivator());
         }
 
-        this.stateBackendName = checkpointStateBackend.getClass().getSimpleName();
+        if (StateBackendLoader.isChangelogStateBackend(checkpointStateBackend)) {
+            this.stateChangelogEnabled = TernaryBoolean.TRUE;
+            StateBackend delegatedStateBackend =
+                    ((DelegatingStateBackend) checkpointStateBackend).getDelegatedStateBackend();
+            this.stateBackendName = delegatedStateBackend.getClass().getSimpleName();
+        } else {
+            this.stateChangelogEnabled = TernaryBoolean.FALSE;
+            this.stateBackendName = checkpointStateBackend.getClass().getSimpleName();
+        }
+
```

Comment:
Can we replace `if` with class-dependent logic here?
For example
```
this.stateBackendName = checkpointStateBackend.getName();
this.stateChangelogEnabled =
        TernaryBoolean.fromBoolean(
                StateBackendLoader.isChangelogStateBackend(checkpointStateBackend));
```

Where `getName()` by default returns `this.getClass().getSimpleName()`,
and changelog returns `delegatedStateBackend.getName()`?

Reply:
Thanks @rkhachatryan , good idea, I have modified as your suggestion.